src/notebooklm/_source_upload.py

```
@@ -313,48 +317,128 @@ async def register_file_source(
+        # set, which makes the probe maximally conservative (any pre-existing
+        # filename match is treated as ambiguous) rather than maximally
+        # permissive. The baseline list is best-effort and never blocks the
+        # primary create path.
         try:
-            result = await rpc_call(
-                RPCMethod.ADD_SOURCE_FILE,
-                params,
-                source_path=f"/notebook/{notebook_id}",
-                allow_null=False,
+            baseline_ids = {source.id for source in await list_sources(notebook_id)}
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)